### PR TITLE
ci: make github workflows run only when relevant files are modified

### LINKED ISSUE #762: ci: make github workflows run only when relevant files are modified
# Description

- Right now, when you modify a file and push it, most of the workflows run. We should run only the ones relevant for each file.

### PR DESCRIPTION
**Pull request by NicolasRampoldi**
# Description

- Makes the CI run workflows relevant to the files modified.
- Adds missing rust tests for ffi bindings.

### CODE REVIEW

**Review by glpecile — APPROVED** [review]

**Review by uri-99 — APPROVED** [review]

**Review by NicolasRampoldi — COMMENTED** [review]
Wait for juli's review

**Review by uri-99 — COMMENTED** [review]
.

**Review by uri-99 — CHANGES_REQUESTED** [review]
I'm pretty sure CI changes affect the PR and don't need to be merged for them to apply

**Review by uri-99 — APPROVED** [review]
was tested by branching and making another PR and worked as expected

File: .github/workflows/build-go.yml [hunk 1]
```diff
@@ -4,8 +4,20 @@ on:
...
     branches: [main]
+    paths:
+      - 'operator/**'
+      - 'aggregator/**'
+      - 'common/**'
+      - 'core/**'
+      - '.github/workflows/build-go.yml'
   pull_request:
     branches: ["*"]
+    paths:
```
Comment (JuArce) [inline .github/workflows/build-go.yml:9]: Missing
- metrics

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -4,7 +4,19 @@ on:
...
       - master
       - main
+    paths:
+      - 'operator/**'
+      - 'aggregator/**'
+      - 'common/**'
+      - 'core/**'
+      - '.github/workflows/golangci-lint.yml'
   pull_request:
+    paths:
```
Comment (JuArce) [inline .github/workflows/golangci-lint.yml:8]: Missing:
- metrics

**Review by JuArce — CHANGES_REQUESTED** [review]

File: .github/workflows/build-go.yml (hunk 1 above)
Comment (NicolasRampoldi) [reply .github/workflows/build-go.yml:9]: Solved in [20793fe](https://github.com/yetanotherco/aligned_layer/pull/763/commits/20793fe4eef1b7d6eb1b59ad662cfae1a6938925)

File: .github/workflows/golangci-lint.yml (hunk 2 above)
Comment (NicolasRampoldi) [reply .github/workflows/golangci-lint.yml:8]: Solved in [20793fe](https://github.com/yetanotherco/aligned_layer/pull/763/commits/20793fe4eef1b7d6eb1b59ad662cfae1a6938925)

**Review by JuArce — APPROVED** [review]

**Pull request merged**